### ISSUE TITLE
Decode level-3 IMX VPU encoded stream.

### ISSUE THREAD

**Issue by UpsilonAudio** [cisco/openh264#2318]
Hello,

I use OpenH264 with Firefox to decode a H264 stream. I have two video encoders, one producing H264 stream with level 4 and one producing a stream with level 3.

When I use the level-4 encoder, there is no problem. But when I use the level-3 encoder, the H264 plugin do not rend the video. Nothing appears on the screen.

Here are the video caps captured with Gstreamer-1.0 :

First encoder - Level 4 (Working) : 
```
"video/x-h264\,\ stream-format\=\(string\)avc\,\ alignment\=\(string\)au\,\ codec_data\=\(buffer\)0142c028ffe1001a6742c028d900780227e5c0440000030004000003007a3c60c92001000468cb8cb2\,\ level\=\(string\)4\,\ profile\=\(string\)constrained-baseline”
```

Second encoder - Level 3 (Not working) : 
```
"video/x-h264\,\ stream-format\=\(string\)avc\,\ alignment\=\(string\)au\,\ codec_data\=\(buffer\)0142401effe100096742401ea680a03d9001000568ce30a480\,\ level\=\(string\)3\,\ profile\=\(string\)constrained-baseline”
```

Nevertheless, if I launch the level-4 encoder, wait for the stream to be decoded by the H264 plugin, stop the encoder, and start the level-3 encoder, then the stream is correctly decoded by the plugin.

In other words, if I directly start the level-3 stream, it does not work. But If I start the level-3 after the level-4 without refreshing the page, then it works.

Do you have any ideas / solutions so that the level-3 stream works directly ?

For information, the level-3 encoder is the IMX VPU Encoder (imxvpuenc_h264 Gstreamer plugin). I also use Janus Gateway with a profile-level-id of `42e082`.

Thank you !

**Comment by jesup** [comment]
We use profile-level-id in negotiation of 42e01f with level-asymmetry-allowed=1.  That means you can send us up to level 3.1, and we can send you up to whatever level you sent to us (or the lowest of the two if you don't do asymmetric level support).  If you send >3.1, it may work, but you shouldn't do it.  

As to why - that depends on what's in the stream, not "level 3" or "level 4", in particular the SPS/PPS.  Just because you told it X doesn't mean it did X...   You need to compare the SPS/PPSs for the two streams

**Comment by UpsilonAudio** [comment]
Thank you for your answer, I will check ASAP the SPS & PPS generated by the encoder. [It seems that](https://github.com/Freescale/gstreamer-imx/issues/17) the encoder I use (imxvpuenc_h264) has problems with the generation of the SPS & PPS. Nevertheless with an other H264 decoder (avdec_h264 Gstreamer plugin) it works well.

**Comment by HaiboZhu** [comment]
No update for long time.
